python/cugraph/cugraph/experimental/dask/centrality/katz_centrality.py

```diff
@@ -0,0 +1,155 @@
+                                       )
+import cugraph.dask.comms.comms as Comms
+import dask_cudf
+import cudf
+import cupy
+
+
+def call_katz_centrality(sID,
+                         data,
+                         graph_properties,
+                         store_transposed,
+                         do_expensive_check,
+                         src_col_name,
+                         dst_col_name,
+                         num_verts,
+                         num_edges,
+                         vertex_partition_offsets,
+                         aggregate_segment_offsets,
+                         alpha,
+                         beta,
+                         max_iter,
+                         tol,
+                         nstart,
+                         normalized):
+    handle = Comms.get_handle(sID)
+    h = ResourceHandle(handle.getHandle())
+    srcs = data[0][src_col_name]
+    dsts = data[0][dst_col_name]
+    weights = cudf.Series(cupy.ones(srcs.size, dtype="float32"))
+
+    if "value" in data[0].columns:
+        weights = data[0]['value']
+
+    initial_hubs_guess_values = None
+    if nstart:
+        initial_hubs_guess_values = nstart["values"]
+
+    mg = MGGraph(h,
+                 graph_properties,
+                 srcs,
+                 dsts,
+                 weights,
+                 store_transposed,
+                 num_edges,
+                 do_expensive_check)
+
+    result = pylibcugraph_katz(h,
+                               mg,
+                               initial_hubs_guess_values,
+                               alpha,
+                               beta,
+                               tol,
+                               max_iter,
+                               do_expensive_check)
+    return result
+
+
+def convert_to_cudf(cp_arrays):
+    """
+    create a cudf DataFrame from cupy arrays
+    """
+    cupy_vertices, cupy_values = cp_arrays
+    df = cudf.DataFrame()
+    df["vertex"] = cupy_vertices
+    df["katz_centrality"] = cupy_values
+    return df
+
+
+def EXPERIMENTAL__katz_centrality(
+    input_graph, alpha=None, beta=1.0, max_iter=100, tol=1.0e-6,
+    nstart=None, normalized=True
+):
+    client = default_client()
+
+    graph_properties = GraphProperties(
+        is_multigraph=False)
+
+    store_transposed = False
+    do_expensive_check = False
+
+    src_col_name = input_graph.renumber_map.renumbered_src_col_name
+    dst_col_name = input_graph.renumber_map.renumbered_dst_col_name
+
+    # FIXME Move this call to the function creating a directed
+    # graph from a dask dataframe because duplicated edges need
+    # to be dropped
+    ddf = input_graph.edgelist.edgelist_df
+    ddf = ddf.map_partitions(
+        lambda df: df.drop_duplicates(subset=[src_col_name, dst_col_name]))
+
+    num_edges = len(ddf)
+    data = get_distributed_data(ddf)
+
+    input_graph.compute_renumber_edge_list(transposed=True)
```

Comment:
Can you add the parameter `legacy_renum_only=True`? This will skip the `C++` renumbering which is causing the hang in the nightly for large scale graph. Furthermore, `renumbering` occurs when creating a `pylibcugraph` graph. Setting `legacy_renum_only=True` will only trigger the `python` renumbering which only renumbers multi column and string vertices which are not yet supported in the C/C++ API